server/middleware/auth.go

```diff
@@ -138,12 +139,26 @@ func AuthFunction(ctx context.Context, jwtValidator *validator.Validator, config
 		return ctx, err
 	}
 
-	validatedToken, err := jwtValidator.ValidateToken(ctx, tkn)
-	if err != nil {
-		log.Warn().Err(err).Msg("Failed to validate access token")
-		return ctx, api.Errorf(api.Code_UNAUTHENTICATED, "Failed to validate access token")
+	validatedToken, found := cache.Get(tkn)
+	if !found {
+		validatedToken, err = jwtValidator.ValidateToken(ctx, tkn)
+		if err != nil {
+			log.Warn().Err(err).Msg("Failed to validate access token")
+			return ctx, api.Errorf(api.Code_UNAUTHENTICATED, "Failed to validate access token")
+		}
+		cache.Add(tkn, validatedToken)
+	} else {
+		// validate expiration
+		validatedClaims, ok := validatedToken.(*validator.ValidatedClaims)
+		if ok {
```

Comment:
This check can be moved inside below `if`.

Reply:
so the else branch is not needed here.

Reply:
updated. thanks